src/Microsoft.DotNet.GitHub.Authentication/GitHubTokenProvider.cs

```diff
@@ -84,8 +99,9 @@ private bool TryGetCachedToken(long installationId, out AccessToken cachedToken)
 
         AccessToken token = _tokenCache[installationId];
 
-        // and update the cache
-        if (token.ExpiresAt.Subtract(DateTimeOffset.UtcNow).TotalMinutes < 15)
+        // If the cached token will expire in less than 30 minutes we won't use it,
+        // Instead GetTokenForInstallationAsync will generate a new one and update the cache
+        if (token.ExpiresAt.UtcDateTime.Subtract(DateTime.Now).TotalMinutes < 15)
```

Comment:
@dkurepa this line

Reply:
Shouldn`t we use `DateTime.UtcNow`, make sure both times are UTC

Reply:
Yes, good catch

Reply:
A bad rebase